clientv3/maintenance.go

```diff
@@ -201,7 +201,12 @@ func (m *maintenance) Snapshot(ctx context.Context) (io.ReadCloser, error) {
 				pw.CloseWithError(err)
 				return
 			}
-			if resp == nil && err == nil {
+			// can "resp == nil && err == nil"
+			// before we receive snapshot SHA digest?
+			// No, server sends EOF with an empty response
+			// after it sends SHA digest at the end
+			// TODO: investigate why this had happened...
+			if resp == nil {
```

Comment:
If we think the code should be unreachable, is it possible to log a warning here?

Reply:
it should be unreachable because if the server sends EOF it will get consumed by pw.CloseWithError(err) which treats EOF as a non-issue i.e. pw.CloseWithError(io.EOF) is basically the same as pw.Close()